sphinx_typo3_theme/__init__.py

```diff
@@ -1,37 +1,38 @@
-import os
+"""Sphinx TYPO3 theme for docs.typo3.org.
+
+"""
+
 import json
-from pkg_resources import get_distribution, DistributionNotFound
-
-try:
-    release = get_distribution('sphinx_typo3_theme').version
-    VERSION = release.split('.')[:3]
-    __version__ = '.'.join(str(v) for v in VERSION)
-    __version_full__ = release
-except DistributionNotFound:
-    VERSION = (0, 0, 0)
-    __version__ = '.'.join(str(v) for v in VERSION)
-    __version_full__ = __version__
+import os
+
+f1path = os.path.join(os.path.dirname(__file__), 'static/theme_info.json' )
+with open(f1path) as f1:
+    theme_info = json.load(f1)
+
+__version__ = theme_info['theme_version_core']
 
 def get_html_theme_path():
-    """Return list of HTML theme paths."""
+    """Return absolute path to parent folder of installed theme."""
+
     return os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
 
-def get_theme_version():
-    """Return the theme version"""
-    return __version__
 
-def _config_inited(app, config):
-    config.html_theme_options['version'] = __version__
-    config.html_theme_options['version_full'] = __version_full__
+def htmlPageContext(app, pagename, templatename, context, doctree):
+    """Inspect reST page metadata possibly select a custom template file.
+
+    A field field list near the top of a reST source file is passed on by
+    Sphinx as file metadata. Here we are looking for a meta field 'template'.
+    If found it specifies the html template file for the page. For example, a
+    line `:template: sitemap.html' right at the beginning of a reST file will
+    tell Sphinx to use the template file `sitemap.html` for this page instead
+    of the default template file `page.html`.
+
+    """
+
+    return app.builder.env.metadata.get(pagename, {}).get('template')
+
 
 def setup(app):
-    app.connect('config-inited', _config_inited)
-    # add_html_theme is new in Sphinx 1.6+
-    if hasattr(app, "add_html_theme"):
-        theme_path = os.path.abspath(os.path.dirname(__file__))
-        app.add_html_theme("sphinx_typo3_theme", theme_path)
```

Comment:
This block is essential, otherwise, the theme cannot be found.

Reply:
Please see 
https://www.sphinx-doc.org/en/1.8/theming.html#distribute-your-theme-as-a-python-package

Reply:
That would bring sphinx back as a requirement, which I want to avoid. Instead put these three lines into `conf.py`. It is ok to append them:
```
html_theme = 'sphinx_typo3_theme'
import sphinx_typo3_theme
html_theme_path = [].append(sphinx_typo3_theme.get_html_theme_path())
```

Reply:
@marble after thinking a bit, I think we can still add the `add_html_theme` function - without needing to add the sphinx requirement. As you can see there is a check if the attribute exists, also means if the attribute does not exist - the only thing that happens is that the theme will not be added.